src/main_macro.rs

```diff
@@ -29,23 +31,68 @@
 ///     println!("{}", x);
 /// });
 /// ```
+///
+/// With command line arguments.
+///
+/// ```rust,ignore
+/// #[macro_use] extern crate quicli;
+/// use quicli::prelude::*;
+/// use std::path::PathBuf;
+///
+/// #[derive(Debug, StructOpt)]
+/// struct Cli {
+///     /// The config file for the cli app.
+///     #[structopt(long = "config", short = "c", parse(from_os_str))]
+///     config: PathBuf,
+/// }
+///
+/// main!(|args: Cli| {
+///     let x = read_file(&args.config)?;
+///     println!("{}", x);
+/// });
+/// ```
+///
+/// With command line arguments including a verbosity flag.
+///
+/// ```rust,ignore
+/// #[macro_use] extern crate quicli;
+/// use quicli::prelude::*;
+/// use std::path::PathBuf;
+///
+/// #[derive(Debug, StructOpt)]
+/// struct Cli {
+///     /// The config file for the cli app.
+///     #[structopt(long = "config", short = "c", parse(from_os_str))]
+///     config: PathBuf,
+///     #[structopt(flatten)]
+///     verbosity: Verbosity,
+/// }
+///
+/// main!(|args: Cli, log_level: Verbosity| {
```

Comment:
This needs to be `verbosity` (lower case), as it refers to the field name, not the type. (I know this is awkward :))